Improve documentation for sensitivity module (refs #3876) (#3877)
GeotrekCE/Geotrek-admin: docs/usage/apis.rst

Les randonnées VTT, trail, vélo et les tours itinérants sont également intégrés dans la passerelle.

Pour plus d'information, se référer à la documentation en ligne de `Sitourisme <https://github.com/GeotrekCE/Sitourisme#sitourisme-paca-api>`_. 

Sensitivity module (or Biodiv'Sports)
-------------------------------------

.. note::

  You can play with API using Biodiv'Sports widget tool: https://biodivsports-widget.lpo-aura.org/

The Geotrek API provides a set of parameters that can be used to filter and sort data. There is a Swagger documentation (see :ref:`advanced-configuration-section` to enable it on your instance if needed) existing to test and browse those parameters that can be find at this address : ``/api/v2/``.

> This section will focus on some common parameters useful to work with sensitivity information and will as give details about some endpoints.

Reviewer: ```suggestion
This section focuses on some common parameters useful to work with sensitivity information and gives details about some endpoints.
```

.. envvar:: Commons parameters

>    If ``language`` parameter is provided, api returns directly translated field, else, a dictionnary of traductions is returned

Reviewer: ```suggestion
   If ``language`` parameter is provided, API returns directly translated fields, else, a dictionnary of traductions is returned
```

      
   e.g. ``/api/v2/sensitivearea_practice/1/?``

   .. code-block:: JSON

      {
         "id":1,
         "name":{
         "fr":"Terrestre",
         "en":"Land",
         "it":null
         }
      }

   e.g. ``/api/v2/sensitivearea_practice/1/?language=en``

   .. code-block:: JSON

      {
         "id":1,
         "name":"Land"
      }

.. envvar:: Sport practices

>    List of outdoor practices

Reviewer: ```suggestion
   List of sport practices
```
Outdoor module is something else in Geotrek

   ``/api/v2/sensitivearea_practice/``

   e.g. https://biodiv-sports.fr/api/v2/sensitivearea_practice/

.. envvar:: Sensitive areas

   List of sensitive areas

   ``/api/v2/sensitivearea/``

   The default output format is ``json``. To obtain output in ``geojson`` format, simply add the ``format=geojson`` parameter.

   ``/api/v2/sensitivearea/?format=geojson`` 

   e.g. https://biodiv-sports.fr/api/v2/sensitivearea/?format=geojson

   **Filtering data**

>    Data can be filtered through those parameters:

Reviewer: ```suggestion
   Data can be filtered through these parameters:
```

   - ``language`` : API language (see :envvar:`Commons parameters`)

   - Expected values: ``fr``, ``en``, ``es`` or ``it``
   - e.g. ``/api/v2/sensitivearea/?language=fr``

>    - ``period`` : Sensitivy period (monthes list)

Reviewer: ```suggestion
   - ``period`` : Sensitivy period (months list)
```

>    - Expected values: List of month number (from 1 to 12), comma separated

Reviewer: ```suggestion
   - Expected values: List of month numbers (from 1 to 12), comma separated
```

   - e.g. ``/api/v2/sensitivearea/?period=4,5,6,7``

>    - ``practices`` : Outdoor sport practices

Reviewer: ```suggestion
   - ``practices`` : Sport practices
```
Outdoor is a dedicated module in Geotrek

   - Expected values: List of practices ids (see :envvar:`Sport practices`)
   - e.g. ``/api/v2/sensitivearea/?practices=1,2``

   - ``structure`` : Organization that declared the sensitive area. 

   - Expected values: List of structures ids
   - e.g. ``/api/v2/sensitivearea/?structures=1,2``

   - ``in_bbox``

   - Expected values: List of bbox coordinates (respectively longitude and latitude South-West then North-East corner), comma separated.
   - e.g. ``/api/v2/sensitivearea/?in_bbox=5.0,45.0,6.0,46.0``

>    full example https://biodiv-sports.fr/api/v2/sensitivearea/?format=geojson&language=fr&practices=1,2&period=4,5,6,7&in_bbox=5.0,45.0,6.0,46.0

Reviewer: ```suggestion
   Full example https://biodiv-sports.fr/api/v2/sensitivearea/?format=geojson&language=fr&practices=1,2&period=4,5,6,7&in_bbox=5.0,45.0,6.0,46.0
```

   **Filtering fields**

>    - ``fields`` : list of expected fields (see :ref:`Field Descriptions <FielDesc>`)

Reviewer: ```suggestion
   - ``fields``: List of expected fields (see :ref:`Field Descriptions <FielDesc>`)
```

   - Expected values: List of field names, comma separated
   - e.g. ``/api/v2/sensitivearea/?fields=name,geometry``

>    - ``omit`` : list of excluded fields (see :ref:`Field Descriptions <FielDesc>`)

Reviewer: ```suggestion
   - ``omit``: List of excluded fields (see :ref:`Field Descriptions <FielDesc>`)
```

   - ``period`` : Area occupancy for each of the 12 months of the year (ordered array of 12 Booleans).
   - ``contact`` : Contact for further information about the sensitive area (string in HTML format).
   - ``practices``: sports practices concerned by the hotspot (array of identifiers).
   - ``info_url`` : URL containing further information about the area (URL).
   - ``structure`` : Structure or acronyme that provided information on the area (string).
   - ``elevation`` : Elevation used to define area sensitivity volume (globally elevation, buffer radius for areas declared as Point).
   - ``geometry`` : Area GeoJSON geometry. Type is always "Polygon".
   - ``species_id``: species identifier or null for regulatory areas.
   - ``kml_url`` : URL of the downloadable KML file representing this regulatory zone.
   - ``openair_url`` : URL of the downloadable OpenAir file representing the regulatory zone (available only for aerial activities).
   - ``attachment`` : List of area attachment files.
   - ``rules`` : List of regulatory rules.
   - ``update_datetime``: last update timestamp.
   - ``create_datetime``: create timestamp.

.. note::

>    Species informations are commons for each species areas share Zones sharing the same ``species_id`` value also share the same values for the ``name``, ``period``, ``practices`` and ``info_url`` fields.

Reviewer: ```suggestion
   Species informations are commons for each species areas sharing the same ``species_id`` value, also share the same values for the ``name``, ``period``, ``practices`` and ``info_url`` fields.
```
